snapcraft/tests/project_loader/grammar/test_to_statement.py

```diff
@@ -0,0 +1,264 @@
+            'body': ['foo'],
+            'else_bodies': [
+                ['bar']
+            ],
+            'target_arch': 'armhf',
+            'expected_packages': {'foo:armhf'}
+        }),
+        ('used else', {
+            'to': 'to armhf',
+            'body': ['foo'],
+            'else_bodies': [
+                ['bar']
+            ],
+            'target_arch': 'i386',
+            'expected_packages': {'bar'}
+        }),
+        ('third else ignored', {
+            'to': 'to armhf',
+            'body': ['foo'],
+            'else_bodies': [
+                ['bar'],
+                ['baz']
+            ],
+            'target_arch': 'i386',
+            'expected_packages': {'bar'}
+        }),
+        ('third else followed', {
+            'to': 'to armhf',
+            'body': ['foo'],
+            'else_bodies': [
+                [{'to armhf': ['bar']}],
+                ['baz']
+            ],
+            'target_arch': 'i386',
+            'expected_packages': {'baz'}
+        }),
+        ('nested armhf', {
+            'to': 'to armhf',
+            'body': [
+                {'to armhf': ['foo']},
+                {'to i386': ['bar']},
+            ],
+            'else_bodies': [],
+            'target_arch': 'armhf',
+            'expected_packages': {'foo:armhf'}
+        }),
+        ('nested i386', {
+            'to': 'to i386',
+            'body': [
+                {'to armhf': ['foo']},
+                {'to i386': ['bar']},
+            ],
+            'else_bodies': [],
+            'target_arch': 'i386',
+            'expected_packages': {'bar:i386'}
+        }),
+        ('nested body ignored else', {
+            'to': 'to armhf',
+            'body': [
+                {'to armhf': ['foo']},
+                {'else': ['bar']},
+            ],
+            'else_bodies': [],
+            'target_arch': 'armhf',
+            'expected_packages': {'foo:armhf'}
+        }),
+        ('nested body used else', {
+            'to': 'to i386',
+            'body': [
+                {'to armhf': ['foo']},
+                {'else': ['bar']},
+            ],
+            'else_bodies': [],
+            'target_arch': 'i386',
+            'expected_packages': {'bar:i386'}
+        }),
+        ('nested else ignored else', {
+            'to': 'to i386',
+            'body': ['foo'],
+            'else_bodies': [
+                [
+                    {'to armhf': ['bar']},
+                    {'else': ['baz']},
+                ],
+            ],
+            'target_arch': 'armhf',
+            'expected_packages': {'bar:armhf'}
+        }),
+        ('nested else used else', {
+            'to': 'to armhf',
+            'body': ['foo'],
+            'else_bodies': [
+                [
+                    {'to armhf': ['bar']},
+                    {'else': ['baz']},
+                ],
+            ],
+            'target_arch': 'i386',
+            'expected_packages': {'baz'}
+        }),
```

Comment:
You're missing a test for ensuring one can still have packages like `name:arch` without the grammar rewriting the requested arch. Please add a test with this in both the body and an else.

Reply:
I just realized I didn't actually respond here: I added two new cases, with "arch specified" in the tag, for that. Good call!